### PR TITLE
Fixes #29356 - Ship puma-plugin-systemd in production

### PR DESCRIPTION
**Pull request by ekohl**
The Puma systemd plugin exposes Puma information to systemd via the sd_notify protocol giving notify, status and watchdog.

RPM packaging PR incoming.

### CODE REVIEW

File: config/puma/production.rb [hunk 1]
```diff
@@ -17,3 +17,5 @@
   dynflow = ::Rails.application.dynflow
   dynflow.initialize! unless dynflow.config.lazy_initialization
 end
+
+plugin :systemd
```
Comment (ehelms) [inline config/puma/production.rb:21]: Will this cause any complications if I attempt to start up production environment in a local development environment?

Comment (ekohl) [reply config/puma/production.rb:21]: As long as the gem is installed, I don't think it should. It checks whether the `NOTIFY_SOCKET` env var is present:
https://github.com/sj26/puma-plugin-systemd/blob/95cd35fa49770e14344ecab0e7800450cc196602/lib/puma/plugin/systemd.rb#L57-L60
https://github.com/sj26/puma-plugin-systemd/blob/95cd35fa49770e14344ecab0e7800450cc196602/lib/puma/plugin/systemd.rb#L182-L187

**Comment by ekohl** [comment]
It looks like this gem is currently limited to `Puma ~> 3.6` but we're on version 4. That's rather annoying and it looks like the gem is not that well maintained.

**Comment by sthirugn** [comment]
> It looks like this gem is currently limited to `Puma ~> 3.6` but we're on version 4. That's rather annoying and it looks like the gem is not that well maintained.

Yeah https://github.com/sj26/puma-plugin-systemd/commits/master shows it was not updated for more than 1.5 years, is this worth using?

**Comment by tbrisker** [comment]
What the status here? shall we close it or do we still want to try to get it to work?

**Comment by ekohl** [comment]
I am considering to fork the plugin because I think it is a good feature to have.

**Comment by ares** [comment]
before the fork, should we try to contact the author and see if we can get merge and gem release permissions? it seems there are some PRs. This [one](https://github.com/sj26/puma-plugin-systemd/pull/2) in particular tells me, someone may be using in other bigger application. Perhaps worth reaching out to @martinpovolny.

Otherwise the PR seems small enough so +1 if the fork is needed.

**Comment by ekohl** [comment]
Looks like there's some repository activity again. I've opened https://github.com/sj26/puma-plugin-systemd/issues/8 to start the discussion around Puma 4 support.

**Comment by tbrisker** [comment]
is there any progress with this @ekohl ?

**Comment by ekohl** [comment]
No, as you can see in the issue there has been no response and I haven't had time to dive into it myself. I still think this should be done though so making it a draft.

**Comment by ares** [comment]
After a discussion with @ekohl, this can be closed for now, another PR replaced it.

**Comment by ekohl** [comment]
For completeness: https://github.com/theforeman/foreman/pull/7923 was merged.

**Pull request closed without merging**